library/connect.c

```diff
@@ -22,13 +22,35 @@ limitations under the License.
 #include "endian_internal.h"
 #include "win32_compat.h"
 
-static const char *TYPE_BIND = "Bind";
-static const char *TYPE_DIAL = "Dial";
 static const char *INVALID_SESSION = "Invalid Session";
 static const int MAX_CONNECT_RETRY = 3;
 
 #define crypto(func) crypto_secretstream_xchacha20poly1305_##func
 
+#define conn_states(XX) \
```

Comment:
these states don't overlap entirely ChannelState. i expect that is that on purpose but it's not clear to me as to why that would be. as long as that's on purpose and i just don't understand 'why' that's fine... if you can add clarity around that in the form of code comments that might be a good idea?

Reply:
it is on purpose, some conn states make no sense for channels: Bound(ing), Accepting, HalfClose, etc.

states are now private to corresponding files, so that should eliminate confusion